src/components/FeaturePanel/renderers/openingHours/complex.ts

```diff
@@ -0,0 +1,53 @@
+import OpeningHours from 'opening_hours';
+import { isInRange } from './utils';
+import { SimpleOpeningHoursTable } from './types';
+
+type Weekday = keyof SimpleOpeningHoursTable;
+const weekdays: Weekday[] = ['su', 'mo', 'tu', 'we', 'th', 'fr', 'sa', 'ph'];
+const weekdayMappings: Record<string, Weekday> = {
+  Sun: 'su',
+  Mon: 'mo',
+  Tue: 'tu',
+  Wed: 'we',
+  Thu: 'th',
+  Fri: 'fr',
+  Sat: 'sa',
+};
+
+const fmtDate = (d: Date) =>
+  d.toLocaleTimeString(undefined, { hour: 'numeric', minute: 'numeric' });
+
+export const parseComplexOpeningHours = (value: string) => {
+  const oh = new OpeningHours(value);
```

Comment:
could you add the `feature.countryCode` here? would it help to find the holidays?

You can get it by `useFeatureContext()` in `<OpeningHoursRenderer>`

Reply:
I though it wouldn't work without a state but I just checked and it works quite good with only the country and coordinates (which are used for sunsets)